packages/core/src/actors.ts

```diff
@@ -0,0 +1,516 @@
+    transition: (state, event, { self, name }) => {
+      if (canceled) {
+        return state;
+      }
+
+      switch (event.type) {
+        case startSignalType:
+          observable = lazyObservable();
+          subscription = observable.subscribe({
+            next: (value) => {
+              self.send({ type: nextEventType, data: value });
+            },
+            error: (err) => {
+              self.send({ type: errorEventType, data: err });
+            },
+            complete: () => {
+              self.send({ type: completeEventType });
+            }
+          });
+          return state;
+        case nextEventType:
+          self._parent?.send(
+            toSCXMLEvent(
+              {
+                type: `xstate.snapshot.${name}`,
+                data: event.data
+              },
+              { origin: self }
+            )
+          );
+          return event.data;
+        case errorEventType:
+          const errorEvent = error(name, event.data);
+          self._parent?.send(
+            toSCXMLEvent(errorEvent, {
+              origin: self
+            })
+          );
+          return state;
+        case completeEventType:
+          self._parent?.send(
+            toSCXMLEvent(doneInvoke(name), {
+              origin: self
+            })
+          );
+          return state;
+        case stopSignalType:
+          canceled = true;
+          subscription?.unsubscribe();
+          return state;
+        default:
+          return state;
+      }
+    },
+    initialState: undefined
+  };
+
+  return behavior;
+}
+
+/**
+ * Creates an event observable behavior that listens to an observable
+ * that delivers event objects.
+ *
+ *
+ * @param lazyObservable A function that creates an observable
+ * @returns An event observable behavior
+ */
+export function fromEventObservable<T extends EventObject>(
+  lazyObservable: Lazy<Subscribable<T>>
+): Behavior<EventObject, T | undefined> {
+  let subscription: Subscription | undefined;
+  let observable: Subscribable<T> | undefined;
+  const nextEventType = Symbol('next');
+  const errorEventType = Symbol('error');
+  const completeEventType = Symbol('complete');
+  let canceled = false;
+
+  // TODO: event types
+  const behavior: Behavior<any, T | undefined> = {
+    transition: (state, event, { self, name }) => {
+      if (canceled) {
+        return state;
+      }
+
+      switch (event.type) {
+        case startSignalType:
+          observable = lazyObservable();
+          subscription = observable.subscribe({
+            next: (value) => {
+              self._parent?.send(toSCXMLEvent(value, { origin: self }));
+            },
+            error: (err) => {
+              self.send({ type: errorEventType, data: err });
+            },
+            complete: () => {
+              self.send({ type: completeEventType });
+            }
+          });
+          return state;
+        case nextEventType:
+          return event.data;
```

Comment:
This is an unused code. It should either be removed or this event should be sent from here:
https://github.com/statelyai/xstate/blob/d3274d1f346c62e6702ecc58b5ff36181a1fbd0b/packages/core/src/actors.ts#L322
and the `self._parent.send(...)` should be performed here